src/main/java/dev/cdevents/CDEventsGenerator.java

```diff
@@ -0,0 +1,129 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class CDEventsGenerator {
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    public static void main(String[] args) {
+        File folder = new File(CDEventConstants.SCHEMA_FOLDER);
+        if (folder.isDirectory()) {
+            File[] files = folder.listFiles((dir, name) -> name.toLowerCase().endsWith(".json"));
+            if (files != null) {
+                //Create Mustache factory and compile event-template.mustache template
+                MustacheFactory mf = new DefaultMustacheFactory();
+                Mustache mustache = mf.compile(CDEventConstants.EVENT_TEMPLATE_MUSTACHE);
+
+                //Generate a class file for each Json schema file using a mustache template
+                for (File file : files) {
+                    SchemaData schemaData = buildCDEventDataFromJsonSchema(file);
+                    generateClassFileFromSchemaData(mustache, schemaData);
+                }
+            }
+        }
+    }
+
+    private static void generateClassFileFromSchemaData(Mustache mustache, SchemaData schemaData) {
+        String classFileName = StringUtils.join(new String[]{schemaData.getCapitalizedSubject(), schemaData.getCapitalizedPredicate(), "CDEvent", ".java"});
+        File classFile = new File("src/main/java/dev/cdevents/events/generated", classFileName);
+        try {
+            FileWriter fileWriter = new FileWriter(classFile);
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+            mustache.execute(bufferedWriter, schemaData).flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new CDEventsException("Exception occurred while generating class file from Json schema ", e);
+        }
+        System.out.println("Rendered event-template has been written to file - "+classFile.getAbsolutePath());
+    }
+
+    private static SchemaData buildCDEventDataFromJsonSchema(File file) {
+        SchemaData schemaData = new SchemaData();
+
+        System.out.println("Processing event JsonSchema file: " + file.getAbsolutePath());
+        try {
+            JsonNode rootNode = objectMapper.readTree(file);
+            JsonNode contextNode = rootNode.get("properties").get("context").get("properties");
+
+            String eventType = contextNode.get("type").get("enum").get(0).asText();
+            System.out.println("eventType: " + eventType);
+            String[] type = eventType.split("\\.");
+            String subject = type[2];
+            String predicate = type[3];
+            String capitalizedSubject = StringUtils.capitalize(subject);
+            if(subject.equals("pipelinerun")){
+                capitalizedSubject = subject.substring(0, 1).toUpperCase() + subject.substring(1, 8) + subject.substring(8, 9).toUpperCase() + subject.substring(9);
```

Comment:
Are you trying to capitalize only char 0 and char 8 in the string?

```suggestion
                capitalizedSubject = StringUtils.capitalize(subject.substring(0, 8)) + 
StringUtils.capitalize(subject.substring(8));
```

If you import a newer version of StringUtils you can capitalize the first character of a substring this way.

Reply:
yes that's right using the StringUtils.capitalize() in the previous line, I can update to this simplified one based on your suggestion,
`capitalizedSubject = capitalizedSubject.substring(0, 8) +
                        StringUtils.capitalize(subject.substring(8));`